src/figuresClasses.ts

```diff
@@ -1,19 +1,114 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+
+function truncation(area: number): number {
+  const TRUNCATION_PRECISION = 100;
 
+  return Math.floor(area * TRUNCATION_PRECISION) / TRUNCATION_PRECISION;
 }
 
-export class Triangle {
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle';
+
+  public color: Color;
+
+  public a: number;
+
+  public b: number;
+
+  public c: number;
+
+  constructor(
+    color : Color,
+    a: number, b: number, c: number,
+  ) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    const sides: number[] = [this.a, this.b, this.c];
+    const longest: number = Math.max(this.a, this.b, this.c);
+
+    sides.splice(sides.indexOf(longest));
+
+    const rest: number = sides.reduce((acc, side) => {
+      return acc + side;
+    }, 0);
+
+    const containsZeroLength: boolean = sides.some((side) => side <= 0);
 
+    if (longest >= rest || containsZeroLength) {
+      throw new Error('Ivalid parameters!');
```

Comment:
The error message 'Ivalid parameters!' has a typo. It should be 'Invalid parameters!'. Always ensure your error messages are clear and free of typos to help with debugging.